crates/hir-def/src/nameres.rs

```diff
@@ -95,6 +95,37 @@ const PREDEFINED_TOOLS: &[SmolStr] = &[
     SmolStr::new_static("rust_analyzer"),
 ];
 
+/// There are some data in the defmap (e.g. extern prelude) that is only needed when analyzing
+/// things in the same crate (and maybe in the IDE layer), e.g. the extern prelude. If we put
+/// it in the DefMap dependant DefMaps will be invalidated when they change (e.g. when we add
+/// a dependency to the crate). Instead we split them out of the DefMap into a LocalDefMap struct.
+/// `crate_local_def_map()` returns both, and `crate_def_map()` returns only the external-relevant
+/// DefMap.
```

Comment:
Small nit: It'd be nice to have a single sentence summary explaining what this does, a space, and then the rest of the discussion. That style makes the rendering of `LocalDefMap` in completions *and* rustdoc nicer.